src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -907,9 +892,11 @@ module T = struct
             ~ledger_proof:res_opt
           |> Deferred.return )
     in
+    let%bind () = yield_result () in
     let%bind coinbase_amount =
-      coinbase_for_blockchain_snark coinbases |> Deferred.return
+      Deferred.return (coinbase_for_blockchain_snark coinbases)
     in
+    let%bind () = yield_result () in
```

Comment:
call yield after the `verify_scan_state_after_apply `  below? That is also taking a significant time. We should probably add a yield call in `Transaction_snark_scan_state.scan_statement` in the next iteration

Reply:
```
...
verify_scan_state_after_apply (skip=false) took $time_elapsed  3237.1082305908203
verify_scan_state_after_apply (skip=false) took $time_elapsed  3296.386480331421
verify_scan_state_after_apply (skip=false) took $time_elapsed  3375.446557998657
verify_scan_state_after_apply (skip=false) took $time_elapsed  2469.630241394043
verify_scan_state_after_apply (skip=false) took $time_elapsed  2484.107494354248
...
```

Reply:
I added some yields to `Statement_scanner.check_invariants` that should help break this up.